SIPS/sip-23.md

```diff
@@ -0,0 +1,673 @@
+  </Box>
+  <Footer>
+    <Button onClick={handleClick}>Click me</Button>
+  </Footer>
+</Container>
+```
+
+#### Copyable
+
+The `Copyable` component renders a text element with a copy button. It accepts a
+`value` prop which is a string to be copied to the clipboard when the button is
+clicked. The component also accepts an optional `sensitive` prop which can be
+used to indicate that the value is sensitive and should be obscured when
+displayed.
+
+```typescript
+type CopyableProps = {
+  value: string;
+  sensitive?: boolean | undefined;
+};
+
+const Copyable: SnapComponent<CopyableProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Copyable value="0x4bbeEB066eD09B7AEd07bF39EEe0460DFa261520" />
+<Copyable value="my-secret-key" sensitive />
+```
+
+#### Divider
+
+The `Divider` component renders a horizontal line. It does not accept any props.
+
+```typescript
+const Divider: SnapComponent;
+```
+
+##### Example
+
+```typescript jsx
+<Divider />
+```
+
+#### Dropdown
+
+The `Dropdown` component renders a dropdown element. It accepts a name prop
+which is a string representing the dropdown name, and a `value` prop which is a
+string representing the selected value. The component also accepts one or more
+option elements, which are rendered as dropdown options.
+
+```typescript
+type DropdownProps = {
+  name: string;
+  value?: string | undefined;
+  children: OptionElement | OptionElement[];
+};
+
+const Dropdown: SnapComponent<DropdownProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Dropdown name="color" value="red">
+  <Option value="red">Red</Option>
+  <Option value="green">Green</Option>
+  <Option value="blue">Blue</Option>
+</Dropdown>
+```
+
+#### Field
+
+The `Field` component renders a field element. It accepts an optional `label`
+prop which is a string representing the field label, and a `children` prop
+which can be an input element, and an optional button element, OR a dropdown
+element. The component also accepts an `error` prop which is a string
+representing an error message.
+
+```typescript
+type FieldProps = {
+  label?: string | undefined;
+  error?: string | undefined;
+  children: [InputElement, ButtonElement] | InputElement | DropdownElement;
+};
+```
+
+##### Example
+
+```typescript jsx
+<Field label="Name">
+  <Input type="text" />
+  <Button onClick={handleClick}>Action</Button>
+</Field>
+```
+
+```typescript jsx
+<Field label="Name">
+  <Dropdown name="color" value="red">
+    <Option value="red">Red</Option>
+    <Option value="green">Green</Option>
+    <Option value="blue">Blue</Option>
+  </Dropdown>
+</Field>
+```
+
+#### Footer
+
+The `Footer` component renders a footer element. It accepts a `children` prop
+which can be one or two buttons.
+
+```typescript
+type FooterProps = {
+  children: ButtonElement | [ButtonElement, ButtonElement];
+};
+
+const Footer: SnapComponent<FooterProps>;
+```
+
+##### Example
+
+```typescript jsx
+<Footer>
+  <Button onClick={handleCancel}>Cancel</Button>
```

Comment:
We need to figure out how to handle this given that @eriknson want to keep the default Cancel button on all dialogs but allow two customizable footer buttons in home pages.

Not sure if it should be part of the spec? 🤔